src/Filter/Filter.php

```diff
@@ -31,27 +31,27 @@ abstract class Filter implements FilterInterface
     protected $value;
 
     /**
-     * @var array
+     * @var array<string, mixed>
      */
     protected $options = [];
 
     /**
-     * @var string
+     * @var string|null
      */
     protected $condition;
 
-    public function initialize($name, array $options = []): void
+    public function initialize(string $name, array $options = []): void
     {
         $this->name = $name;
         $this->setOptions($options);
     }
 
-    public function getName()
+    public function getName(): ?string
```

Comment:
Name is never null isnt it ?

Reply:
Fixed

Reply:
It is null when the object is construct. See my last comment on this PR